### PR TITLE
Eliminating the NPE when using statistics file

### LINKED ISSUE #1464: Saving statistics fails with NPE
I don't really know the circumstances but if the statistics path is `null` then a NPE is thrown and is not caught in the higher layer. Tomcat logs this exception afterwards.

```log
10-Mar-2017 14:01:11.392 SEVERE [http-nio-8080-exec-4] org.apache.catalina.core.StandardContext.listenerStop Exception sending context destroyed event to listener instance of class org.opensolaris.opengrok.web.WebappListener
 java.lang.NullPointerException
	at java.io.File.<init>(File.java:277)
	at org.opensolaris.opengrok.configuration.RuntimeEnvironment.saveStatistics(RuntimeEnvironment.java:1474)
	at org.opensolaris.opengrok.web.WebappListener.contextDestroyed(WebappListener.java:137)
	at org.apache.catalina.core.StandardContext.listenerStop(StandardContext.java:4776)
	at org.apache.catalina.core.StandardContext.stopInternal(StandardContext.java:5390)
	at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:232)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:160)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:725)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:701)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:717)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:586)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:460)
	at org.apache.catalina.startup.HostConfig.check(HostConfig.java:1556)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tomcat.util.modeler.BaseModelMBean.invoke(BaseModelMBean.java:300)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
	at org.apache.catalina.manager.ManagerServlet.check(ManagerServlet.java:1460)
	at org.apache.catalina.manager.ManagerServlet.deploy(ManagerServlet.java:906)
	at org.apache.catalina.manager.ManagerServlet.doGet(ManagerServlet.java:344)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:622)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.filters.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:108)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:614)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:518)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1091)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:673)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1526)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1482)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

### PR DESCRIPTION
**Pull request by tulinkry**
fixes #1464

### CODE REVIEW

File: test/org/opensolaris/opengrok/configuration/RuntimeEnvironmentTest.java [hunk 1]
```diff
@@ -510,9 +510,9 @@ public void testCanAcceptMessage() throws Exception {
         Assert.assertFalse(instance.canAcceptMessage(m1));
         m1.setExpiration(new Date(System.currentTimeMillis() - 1));
         Assert.assertFalse(instance.canAcceptMessage(m1));
-        m1.setExpiration(new Date(System.currentTimeMillis() - 0));
+        m1.setExpiration(new Date(System.currentTimeMillis() + 50));
         Assert.assertTrue(instance.canAcceptMessage(m1));
-        m1.setExpiration(new Date(System.currentTimeMillis() + 1));
+        m1.setExpiration(new Date(System.currentTimeMillis() + 100));
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/configuration/RuntimeEnvironmentTest.java]: Appeared as a too little marging for testing `isExpired` when running just few tests (little load).

File: test/org/opensolaris/opengrok/web/StatisticsTest.java [hunk 2]
```diff
@@ -94,28 +94,6 @@ public void testMaybeRefresh() {
...
-    public void testAddRequest_HttpServletRequest_String() {
-        System.out.println("addRequest");
-        HttpServletRequest req = null;
-        String category = "";
-        Statistics instance = new Statistics();
-        instance.addRequest(req, category);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
-    }
-     */
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/web/StatisticsTest.java]: There was this auto generated test section from NB which I think I forgot to remove last time.

File: test/org/opensolaris/opengrok/configuration/RuntimeEnvironmentTest.java (hunk 1 above)
Comment (tulinkry) [inline test/org/opensolaris/opengrok/configuration/RuntimeEnvironmentTest.java]: #1470

**Comment by tarzanek** [comment]
merging, thank you

**Pull request merged**